### PR TITLE
Use more exact name match when deleting static routes to HO nodes

### PR DESCRIPTION
**Pull request by pliurh**
#### What this PR does and why is it needed
It can help to avoid the unintended match when the '-gr' substring exists in a node name.

#### Which issue(s) this PR fixes

Fixes https://issues.redhat.com/browse/OCPBUGS-37855

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by kyrtapz** [comment]
Unit tests are failing:
```
2024-08-06T06:34:30.1724651Z [91m[1m• Failure [2.555 seconds][0m
2024-08-06T06:34:30.1725016Z Hybrid SDN Master Operations
2024-08-06T06:34:30.1725694Z [90m/home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/ovn/hybrid_test.go:132[0m
2024-08-06T06:34:30.1726512Z   [91m[1mhandles a HO node is switched to a OVN node [It][0m
2024-08-06T06:34:30.1727366Z   [90m/home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/ovn/hybrid_test.go:1216[0m
2024-08-06T06:34:30.1727876Z 
2024-08-06T06:34:30.1728047Z   [91mTimed out after 2.000s.
2024-08-06T06:34:30.1728349Z   Expected
2024-08-06T06:34:30.1728735Z       <[]*nbdb.LogicalRouterStaticRoute | len:2, cap:2>: [
2024-08-06T06:34:30.1729156Z           {
2024-08-06T06:34:30.1729645Z               UUID: "45395d42-abfa-449b-8dbb-3643880c2e3b",
2024-08-06T06:34:30.1730069Z               BFD: nil,
2024-08-06T06:34:30.1730578Z               ExternalIDs: {
2024-08-06T06:34:30.1731172Z                   "name": "hybrid-subnet-node1:node-windows",
2024-08-06T06:34:30.1731583Z               },
2024-08-06T06:34:30.1731919Z               IPPrefix: "10.1.3.0/24",
2024-08-06T06:34:30.1732317Z               Nexthop: "10.1.1.3",
2024-08-06T06:34:30.1732683Z               Options: nil,
2024-08-06T06:34:30.1733031Z               OutputPort: nil,
2024-08-06T06:34:30.1733383Z               Policy: nil,
2024-08-06T06:34:30.1733806Z               RouteTable: "",
2024-08-06T06:34:30.1734096Z           },
2024-08-06T06:34:30.1734331Z           {
2024-08-06T06:34:30.1734806Z               UUID: "927fb276-ded3-45ee-b7dc-7d87374e4b2a",
2024-08-06T06:34:30.1735227Z               BFD: nil,
2024-08-06T06:34:30.1735544Z               ExternalIDs: {
2024-08-06T06:34:30.1736056Z                   "name": "hybrid-subnet-node1-gr",
2024-08-06T06:34:30.1736713Z               },
2024-08-06T06:34:30.1737057Z               IPPrefix: "10.1.3.0/24",
2024-08-06T06:34:30.1737718Z               Nexthop: "100.64.0.1",
2024-08-06T06:34:30.1738361Z               Options: nil,
2024-08-06T06:34:30.1738954Z               OutputPort: nil,
2024-08-06T06:34:30.1739543Z               Policy: nil,
2024-08-06T06:34:30.1740114Z               RouteTable: "",
2024-08-06T06:34:30.1740601Z           },
2024-08-06T06:34:30.1740933Z       ]
2024-08-06T06:34:30.1741346Z   to have length 0[0m
```

File: go-controller/pkg/ovn/hybrid_test.go [hunk 1]
```diff
@@ -1243,6 +1243,7 @@ var _ = ginkgo.Describe("Hybrid SDN Master Operations", func() {
 				DnatSnatIP:           "169.254.0.1",
 			}
 			testNode := node1.k8sNode("2")
+			testNode.Annotations[hotypes.HybridOverlayDRIP] = nodeHOIP
```
Comment (trozet) [inline go-controller/pkg/ovn/hybrid_test.go:1246]: why this change?

**Comment by trozet** [comment]
Note related unit test flake I'm fixing: https://github.com/ovn-org/ovn-kubernetes/pull/4583

File: go-controller/pkg/ovn/hybrid_test.go (hunk 1 above)
Comment (trozet) [reply go-controller/pkg/ovn/hybrid_test.go:1246]: I think you did this to try to make the unit test pass, but it is flawed anyway. Will be fixed by https://github.com/ovn-org/ovn-kubernetes/pull/4583/files

Remove this change and the rest of your PR lgtm

**Review by trozet — APPROVED** [review]
